src/components/ShortyString/ShortyString.tsx

```diff
@@ -27,12 +27,21 @@ export default function ShortyString({
     collapseLabel = 'Show less',
 }: Props) {
     const [expanded, setExpanded] = React.useState(false);
-    const hasToggle = value.length > limit;
-    const length =
-        displayLength && !expanded ? `(${value.length} symbols)` : undefined;
 
-    const text = expanded || value.length <= limit ? value : value.slice(0, limit - 4) + '\u00a0...';
-    const label = expanded ? collapseLabel : expandLabel;
+    const toggleLabelAction = expanded ? collapseLabel : expandLabel;
+    const toggleLabelSymbolsCount = displayLength && !expanded
+        ? ` (${value.length} symbols)`
+        : '';
+    const toggleLabel = toggleLabelAction + toggleLabelSymbolsCount;
+
+    // showing toogle button with a label that is longer than the hidden part is pointless,
+    // hence compare to limit + length
+    const hasToggle = value.length > limit + toggleLabel.length;
```

Comment:
toggleLabel length should also include 4 additional symbols (three dots and a space, they are counted in slice function below). Besides, length of margin (2 em) between text and button could be considered here as well

Reply:
They are included in the limit. The final string is `text - 4 chars + space + 3 dots` which length equals to `limit`.

Considering the margin is an overkill, the point of this change is not to trim the text if the toggle label is longer than the hidden part.